### PR TITLE
User gudie update based on the new way of dependency handling.

### PR DESCRIPTION
**Pull request by bruntib**

### CODE REVIEW

File: doc/deps.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+cd odb-2.4.0.tar.gz
+./configure --prefix=<odb_install_dir>
+make install
+cd ..
+
+export CMAKE_PREFIX_PATH=<odb_install_dir>:$CMAKE_PREFIX_PATH
+```
+
+# Build CodeCompass
+
```
Comment (csordasmarton) [inline doc/deps.md]: We have to export the installation lib directory of thrift, odb, and clang for running CodeCompass:
```bash
export LD_LIBRARY_PATH=<thrift_install_dir>/lib:<clang_install_dir>/lib:<odb_install_dir>/lib:$CMAKE_PREFIX_PATH
```

File: doc/usage.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+stopped and restarted with an existing database, so the parsing can continue
+from where it stopped. **Furthermore it sets some environment variables which
+are necessary to start the `CodeCompass_parser` binary, so its usage is
+mandatory!**
+
+### Using example:
+
+```bash
+# Parse the prepared project
+keepalive CodeCompass_parser \
```
Comment (csordasmarton) [inline doc/usage.md]: The project name flag (`-n`) is also necessary.
Furthermore, I think this should be a real example like this:
```bash
# Parse an example C++ project. We suppose that the database is located on port 6250 on localhost
keepalive CodeCompass_parser \
  -d "pgsql:host=localhost;database=example;user=myusername;port=6250;" 
  -w ~/cc/workdir \
  -n example \
  -i ~/cc/projects/example/compile_commands.json \
  -j4
```

File: doc/usage.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  config and log files are located. 
+- **Port**: Port number of the webserver to listen on.
+- **Database**: The plugins can use an SQL database as storage. By the
+  connection string the user can give the location of a running database
+  system. 
+
+### Using example:
+
+```bash
+keepalive CodeCompass_webserver \
```
Comment (csordasmarton) [inline doc/usage.md]: It should be also a real example:
```bash
# Start the server listening on port "6251". We suppose that database is located on port "6250".
keepalive CodeCompass_webserver \
  -d "pgsql:host=localhost;database=example;user=myusername;port=6250;" \
  -w ~/cc/workdir \
  -p 6251

# Open up a browser and access "localhost:6251"
```

File: doc/usage.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+# CodeCompass usage
+
+## Setup database (one time only)
```
Comment (csordasmarton) [inline doc/usage.md]: I think we should write a using example at this section too:
### Using example:
```bash
mkdir -p /path/to/pgsql_data
initdb -D /path/to/pgsql_data -E "SQL_ASCII"

# Start PostgreSQL server on port 6250
postgres -D /path/to/pgsql_data -p 6250
```

File: doc/deps.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+- **openjdk-8-jdk**: For search parsing CodeCompass uses an indexer written in
+  Java.
+- **libssl-dev**: OpenSSL libs are required by Thrift.
+- **libsqlite3-dev**: SQLite can be used for persistence. This is needed only if
+  you choose SQLite as a database system.
+- **libgraphviz-dev**: GraphViz is used for generating diagram visualitaions.
+- **libmagic-dev**: For detecting file types.
+- **libgit2-dev**: For compiling Git plugin in CodeCompass.
+- **npm**: For handling JavaScript dependencies for CdoeCompass web GUI.
+- **ctags**: For search parsing.
```
Comment (csordasmarton) [inline doc/deps.md]: `gtest` is also necessary for building CodeCompass.

File: doc/usage.md (hunk 4 above)
Comment (bruntib) [inline doc/usage.md]: This is written in the linked documentations.

Comment (csordasmarton) [inline doc/usage.md]: Yes, I know, but we should write a using example at the bottom of this section like I wrote above.

**Pull request merged**